include/datafile/DataFileUpgrade.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * DataFileUpgrade.h - base functor for DataFile upgrade routines
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#pragma once
+
+#ifndef LMMS_DATAFILEUPGRADE_H
+#define LMMS_DATAFILEUPGRADE_H
+
+#include "DataFile.h"
+
+#include "lmms_basics.h"
+
+namespace lmms
+{
+
+
+// Base functor with some helper methods
+class DataFileUpgrade
+{
+public:
+	DataFileUpgrade(DataFile& document) : m_document(document) {}
+
+	virtual void upgrade() {}
+
+	void operator()() { upgrade(); }
+
+	// wrappers
+	QDomNodeList elementsByTagName( const QString& tagName );
+	QDomElement createElement(const QString& tagName);
+	QDomElement documentElement() const;
+	QDomElement firstChildElement(const QString& tagName) const;
+	QDomElement firstChildElement() const;
```

Comment:
Since you provided the definitions in the header file, I would inline most of these functions here.

Reply:
I'm not sure what you mean here. These functions are declared here, but their bodies are not defined, and as they are member functions in a class body they are implicitly inline.

Reply:
> I'm not sure what you mean here. These functions are declared here, but their bodies are not defined, and as they are member functions in a class body they are implicitly inline.

I meant to say add an `inline` specifier, but since it's redundant, this can be ignored too :+1:.